### PR TITLE
Buffs lightning holoparasite to not be total shit

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: Lightning Holoparasites will actually shock relatively often instead of 'sometimes, maybe, if you're really lucky'
/:cl:

Splits holoparasite types up into separate files.
Adds two new admin/lucky miner-only holoparasite types; Protector and Charger.
Protector parasites have very high damage resist, a decent attack, and leash the summoner to them instead of leashing to the summoner.
Charger parasites move fast and can charge at a location, damaging and stunning the first mob hit, if any.
charger

### CODE REVIEW

File: code/modules/mob/living/simple_animal/guardian/types/lightning.dm [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+						C.stuttering += 1
+						spawn(20)
+							if(C)
+								C.jitteriness = max(C.jitteriness - 990, 10)
+					L.visible_message(
+						"<span class='danger'>[L] was shocked by the lightning chain!</span>", \
+						"<span class='userdanger'>You are shocked by the lightning chain!</span>", \
+						"<span class='italics'>You hear a heavy electrical crack.</span>" \
+					)
+				L.adjustFireLoss(0.7) //adds up very rapidly
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/guardian/types/lightning.dm]: No, really it does. One hit on an enemy/one enemy chain did about 24 damage before it disappeared.

**Comment by KorPhaeron** [comment]
>Charger parasites move fast and can charge at a location, damaging and stunning the first mob hit, if any.

Adding this to juggernauts instead of laser immunity when?

**Comment by Robustin** [comment]
I'd probably prefer laser bounces to turbojuggs.

File: code/game/objects/effects/overlays.dm [hunk 2]
```diff
@@ -73,6 +73,21 @@
...
+/obj/effect/overlay/temp/decoy/New(loc, atom/mimiced_atom)
+	..()
+	alpha = initial(alpha)
+	if(mimiced_atom)
+		icon = mimiced_atom.icon
+		icon_state = mimiced_atom.icon_state
+		dir = mimiced_atom.dir
+	spawn(0)
+		while(alpha)
+			alpha -= initial(alpha) / duration //reduce alpha over duration
```
Comment (RemieRichards) [inline code/game/objects/effects/overlays.dm]: why are these 4 lines not just:
```animate(src, alpha = 0, time = initial(alpha))```?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -1,3 +1,6 @@
+
+var/list/parasites = list() //all currently existing/living guardians
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: ```var/global/list/parasites = list()```

**Pull request merged**